app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -204,6 +206,10 @@ public void reportProgress(int progress) {
                 //Either way, don't re-do this step
                 this.publishProgress(PROGRESS_CLEANED);
 
+                if (isCancelled()) {
+                    // avoid making the http request if user cancelled the task
+                    return PullTaskResult.UNKNOWN_FAILURE;

```

Comment:
Does this response just not get used? Seems like adding a cancelled result wouldn't be terrible, but if I recall this doesn't go anywhere if the task was cancelled?